src/WebJobs.Script/Rpc/WebHostLanguageWorkerChannelManager.cs

```diff
@@ -133,13 +135,19 @@ public Task<bool> ShutdownChannelIfExistsAsync(string language, string workerId)
                             ILanguageWorkerChannel workerChannel = channelTask.Result;
                             if (workerChannel != null)
                             {
+                                while (channelTask.Result.FunctionInputBuffers.Values.Sum(val => val.Count) > 0 && timer.ElapsedMilliseconds < _disposeGraceTimeInMilliseconds)
```

Comment:
this wait should only kick in when RpcServer is shutting down. If not, this will block when host is intentionally disposing worker channels, for example after specialization.

Reply:
But in cases where the host is intentionally shutting it down, can the buffer be non-empty? If the buffer is empty, this while loop basically becomes a no-op..